### DISCUSSION TITLE
某些段落之间平白出现一空行

### DISCUSSION THREAD

**Discussion by earthmoon** [whutug/whu-thesis#285] [Q&A]
您好，
在生成pdf时，如果一个页面只有段落文字，而无标题，那么这个页面的段落之间就会出现空行。这是怎么回事呢？
谢谢！

**Comment by SwitWu** [comment]
可以看看 #276 中的讨论

**Reply by earthmoon** [reply]
明白了，为了保证页面底部一致，所以在段落之间插入空行对吧。

**Reply by earthmoon** [reply]
但是还是有个问题。比如一页中出现三段，其中每段之间出现空行，就是两个空行。这种情况，不应该补一行文字，再将页面纵向的雨量分给每段之间的间距吗？

**Reply by SwitWu** [reply]
也不能叫做空行，在 TeX 中，相邻段落之间会插入一个称为 `\parskip` 的垂直粘连，在 plain TeX 中，`\parskip` 的默认值为 0pt plus 1pt，也就是说这个粘连的自然长度为零，并且有 1pt 的伸展能力，但是没有收缩能力。但是注意，这个 1pt 的伸展能力并不是说最多只能伸展 1pt，而是建议最大伸展值为 1pt。

若想了解更多，可以看看 TeXbook Chapter 12 (Glue), Chapter 14 (How TeX Breaks Paragraphs into Lines)

**Reply by earthmoon** [reply]
感觉如果能在底部对齐的情况下，尽可能多塞行，再把竖直方向的余量匀给每行，这样看起来可能不会那么明显。

**Reply by SwitWu** [reply]
这和 TeX 的断行分页算法有关

**Reply by earthmoon** [reply]
嗯嗯。我都没注意到以前的版本中，底部并没有对齐。
